lilio/utils.py

```diff
@@ -115,6 +116,9 @@ def infer_input_data_freq(
         if data_freq is None:  # Manually infer the frequency
             data_freq = (data.time.values[1:] - data.time.values[:-1]).min()
 
+    if "ME" or "MS" in data_freq:
+        data_freq = data_freq.replace("ME", "M").replace("MS", "M")
```

Comment:
Do note that this resolves to
```
(if "ME") or (if "MS" in data_freq)
```
and `if "blabla"` resolves to True. Anything but 0, False, None and empty iterators resolve to True.

But the whole if statement here is not needed, you can move it down to be after the `if isinstance(data_freq, str):`.